plugins/installer/override/override.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  Overrides.Override
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\Factory;
+
+/**
+ * Override Plugin
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PlgInstallerOverride extends CMSPlugin
+{
+
+	/**
+	 * Application object.
+	 *
+	 */
+	protected $app;
+
+	/**
+	 * Load the language file on instantiation.
+	 *
+	 * @var    boolean
+	 *
+	 * @since  3.6.0
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Method to get com_templates model instance.
+	 *
+	 * @param   string  $name    The model name. Optional
+	 * @param   string  $prefix  The class prefix. Optional
+	 *
+	 * @return  \Joomla\CMS\MVC\Model\BaseDatabaseModel The model.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getModel($name = 'Template', $prefix = 'Administrator')
+	{
+		$app = Factory::getApplication();
+		$model = $app->bootComponent('com_templates')->createMVCFactory($app)->createModel($name, $prefix);
+
+		return $model;
+	}
+
+	/**
+	 * Method to prepare changed or updated core file.
+	 *
+	 * @return   array  A list of changed files.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getUpdatedFiles()
+	{
+		$after  = $this->afterEventFiles;
+		$before = $this->beforeEventFiles;
+		$size1  = count($after);
+		$size2  = count($before);
+
+		$result = null;
+
+		if ($size1 === $size2)
+		{
+			$result = array();
+
+			for ($i = 0; $i <= $size1; $i++)
+			{
+				if ($after[$i]->coreFile !== $before[$i]->coreFile)
+				{
+					$result[] = $after[$i];
+				}
+			}
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Method to get core list of override files.
+	 *
+	 * @return   array  The list of core files.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function getOverrideCoreList()
+	{
+		// Get template model
+		$templateModel = $this->getModel();
+		$result = $templateModel->getCoreList();
+
+		return $result;
+	}
+
+	/**
+	 * Event before extension update.
+	 *
+	 * @return   void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onExtensionBeforeUpdate()
+	{
+		$this->beforeEventFiles = $this->getOverrideCoreList();
+	}
+
+	/**
+	 * Event after extension update.
+	 *
+	 * @return   void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onExtensionAfterUpdate()
```

Comment:
The methods `onExtensionAfterUpdate()`, `onJoomlaAfterUpdate()`; and `onInstallerAfterInstaller()`  have much similar code. 
Maybe you can put the duplicated code in one method?

Reply:
@astridx Please, don't worry about it. This is only a test, of the plugin to make sure all methods are working fine. In, my next PR I will clean up these things with more functionality  :)

Reply:
OK. That is fine. Then we work on that in the next PR.